### ISSUE TITLE
Disentangling Correlated Speaker and Noise for Speech Synthesis via Data Augmentation and Adversarial Factorization

### ISSUE THREAD

**Issue by patrick-g-zhang** [patrick-g-zhang/papers-i-read#1]
# Abstract

- a conditional generative model with factorized latent variables
-  using data augmentation to add noise that is not correlated with speaker identity and whose
label is known during training
-  using adversarial factorization to improve disentanglement

# Details
## Hinders and Problems
- difficult to annotate background noise label like speaker 
- speaker identity has strong correlation with recording conditions 

## Main contributions
- a conditional generative model with factorized latent variables to control different attributes
- data augmentation by adding background noise to training utterances in order to counteract the inherent speaker-noise correlation and to create ground truth noisy acoustic condition labels
-  adversarial training based on the generated labels to encourage disentanglement between latent variables.

## Conditional Generative Model

![img](https://dl.dropbox.com/s/7xgbbevs6l4jaa7/Screenshot%202020-06-26%2013.38.19.png?dl=0)
Speaker Classifier encourages information that is discriminative  between speakers to be encoded in speaker encoding.

### Speaker invariant data augmentation
 Acoustic conditions are correlated with speaker. Decorrelating this factor by leveraging prior knowledge that adding noise should not affect speaker identity.    
HOW TO？ **augment the original training set with a noisy copy that mixes each utterance with
a randomly selected piece of background noise at a randomly sampled signal-to-noise ratio (SNR),
but reuses the same transcript and speaker label as the original utterance**（make SNR and types of noise less discriminative to speakers ）

### Augmentation-adversarial training
- binary augmentation classifier discourages ![equation](https://latex.codecogs.com/gif.latex?z_s) from encoding acoustic condition information
-  use augmentation label (original/augmented) to replace the acoustic condition label (clean/noisy)
- using DANN(Domain adversarial Neural network)

### Training objective 
- synthesizer:![equation](https://latex.codecogs.com/gif.latex?p(speech|z_s,&space;z_r,&space;text))
- two encoders: ![equation](https://latex.codecogs.com/gif.latex?q(z_s|speech),q(z_r|speech))
- speaker and augmentation classifier: ![equation](https://latex.codecogs.com/gif.latex?p(y_s|z_s),p(y_r|z_r))
- maximize conditional likelihood and  ![equation](https://latex.codecogs.com/gif.latex?z_s)  contains about speakers, minimizing the H-divergence between the ![equation](https://latex.codecogs.com/gif.latex?z_s) inferred from the original utterances and that from the augmented ones.
-  gradient reversal layer is inserted after the input to the augmentation classifier
![img](https://dl.dropbox.com/s/2eac4359tcwprg1/Screenshot%202020-06-27%2003.07.22.png?dl=0)

### Dataset
- generated a noisy speech dataset with correlated speaker and noise conditions using the VCTK corpus and background noise from the CHiME-4 challenge
- speaker correlated noise; half of speakers each of speaker with a noise
- augmented set

### Latent space disentanglement
-  quantify the degree of disentanglement by training speaker and noise classifiers on zs and zr
- second line of table(without adversarial classifier), part of noise information leak to zs
- Without data augmentation and adversarial training, failed to disentangle speaker from noise(speaker and noise are correlated)
![img](https://dl.dropbox.com/s/5h5ddxeaaoogu19/Screenshot%202020-06-27%2005.27.19.png?dl=0)

### Evaluation of synthesized speech
- Control of acoustic conditions
- Control of speaker identity
- Subjective naturalness evaluation
- Hyperparameter sensitivity

## Personal Ideas
- Interesting performance of system when speaker and noise are highly correlated.
- give an approach to decorrelate two attributes
- alternative to speaker embedding with speaker encoder + classifier
